Haskell/Cabal.sublime-syntax

```diff
@@ -0,0 +1,406 @@
+%YAML 1.2
+---
+# https://cabal.readthedocs.io/en/3.4/cabal-package.html
+# https://www.sublimetext.com/docs/syntax.html
+name: Cabal
+scope: source.cabal
+version: 2
+
+file_extensions:
+  - cabal
+  - cabal.project
+
+###############################################################################
+
+variables:
+  identifier: (?:[[:alpha:]][-_[:alnum:]]*)
+  module_identifier: (?:[[:upper:]][\w']*)
+
+  section_tags: |-
+    (?x:
+      common | executable | benchmark | custom-setup | flag | library | package
+    | source-repository | test-suite
+    )
+
+  boolean_tags: |-
+    (?x: benchmarks | default | optimization | tests )
+  compiler_tags: |-
+    (?x: tested-with )
+  dependency_tags: |-
+    (?x: (?:{{identifier}}-)? depends )
+  module_tags: |-
+    (?x: (?:{{identifier}}-)? modules )
+  option_tags: |-
+    (?x: (?:{{identifier}}-)? options | flags )
+  path_tags: |-
+    (?x: (?:{{identifier}}-)? (?: dir | file | path ) s? | main-is | packages )
+  url_tags: |-
+    (?x: bug-reports | homepage | location | package-url )
+  version_tags: |-
+    (?x: cabal-version | version )
+
+  compiler_values: |-
+    (?xi: GHC | UHC | LHC )
+
+  escape_chars: |-
+    (?x:
+      NUL | SOH | STX | ETX | EOT | ENQ | ACK | BEL | BS | HT | LF | VT | FF
+    | CR | SO | SI | DLE | DC1 | DC2 | DC3 | DC4 | NAK | SYN | ETB | CAN | EM
+    | SUB | ESC | FS | GS | RS | US | SP | DEL
+    | [abfnrtv\\\"'\&]
+    )
+  escape_sequence: |-
+    (?x:
+        (\\{{escape_chars}}) # Escapes
+      | (\\[0-9]+)           # Decimal Escapes
+      | (\\o[0-7]+)          # Octal Escapes
+      | (\\x[0-9A-Fa-f]+)    # Hexadecimal Escapes
+      | (\^[A-Z@\[\]\\\^_])  # Control Chars
+    )
+
+###############################################################################
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: sections
+    # mappings
+    - include: boolean-keys
+    - include: compiler-keys
+    - include: dependency-keys
+    - include: module-keys
+    - include: option-keys
+    - include: path-keys
+    - include: url-keys
+    - include: version-keys
+    - include: other-keys
+    # expressions
+    - include: function-calls
+    - include: keywords
+    - include: operators
+
+###[ COMMENTS ]################################################################
+
+  comments:
+    - match: (--+).*\n?
+      scope: comment.line.double-dash.cabal
+      captures:
+        1: punctuation.definition.comment.cabal
+
+###[ SECTIONS ]################################################################
+
+  sections:
+    - match: ^(({{section_tags}})(?:\s+([^\s,]+))?)\s*$
+      captures:
+        1: meta.toc-list.sections.cabal
+        2: keyword.declaration.section.cabal
+        3: entity.name.section.cabal
+
+###[ BOOLEANS ]################################################################
+
+  boolean-keys:
+    - match: ^(\s*)({{boolean_tags}})(:)
+      captures:
+        2: meta.mapping.key.cabal entity.name.tag.cabal
```

Comment:
I've been out of RFC discussions for a while, but did we reach an agreement on how to scope keys?

Reply:
No, I am actually not very confident of how to scope these things in such config files. But as CSS and HTML make use of `entity.name` or `entity.other` heavily it was the first thing which came to my mind here.

It also matched a recent description Will posted about tags, which was something like "A tag is a token which denotes a start of a block of text" or something along these lines.

I am happy with any better alternative.